src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,34 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="ConstructorParameterShouldMatchFieldName" xml:space="preserve">
+    <value>For proper deserialization, change the name of parameter '{1}' in the constructor of '{0}' to match the bound field '{2}'</value>
+  </data>
+  <data name="ConstructorParameterShouldMatchPropertyName" xml:space="preserve">
+    <value>For proper deserialization, change the name of parameter '{1}' in the constructor of '{0}' to match the bound property '{2}'</value>
+  </data>
+  <data name="ConstructorParameterShouldMatchPropertyWithPublicVisibility" xml:space="preserve">
+    <value>For proper deserialization, change the visibility of the bound property '{2}' to public</value>
+  </data>
+  <data name="ConstructorParametersShouldMatchPropertyNamesDescription" xml:space="preserve">
+    <value>For a constructor having [JsonConstructor] attribute each parameter name must match with a property or field name for proper deserialization.</value>
```

Comment:
Seems it doesn't need to have 4 description for each case, they are very repetitive, I would prefer only one, like the title. Or at least 2 versions would be enough.
```suggestion
    <value>For a constructor having [JsonConstructor] attribute each parameter name must match with a public property or field name for proper deserialization.</value>
```